Bricks/cerealize/cerealize.h

```diff
@@ -44,72 +44,29 @@ SOFTWARE.
 
 #include "../3party/cereal/include/external/base64.hpp"
 
+#include "../strings/is_string_type.h"
+
 #include "../rtti/dispatcher.h"
 #include "../template/decay.h"
 
 namespace bricks {
 namespace cerealize {
 
-// Helper compile-type test to tell string-like types from cerealizable types.
-template <typename T>
-struct is_string_type_impl {
-  constexpr static bool value = false;
-};
-template <>
-struct is_string_type_impl<std::string> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<char> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<std::vector<char>> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<std::vector<int8_t>> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<std::vector<uint8_t>> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<char*> {
-  constexpr static bool value = true;
-};
-template <size_t N>
-struct is_string_type_impl<char[N]> {
-  constexpr static bool value = true;
-};
-template <>
-struct is_string_type_impl<const char*> {
-  constexpr static bool value = true;
-};
-template <size_t N>
-struct is_string_type_impl<const char[N]> {
-  constexpr static bool value = true;
-};
-// Microsoft Visual Studio compiler is strict with overloads,
-// explicitly exclude string-related types from cereal-based implementations.
-template <typename TOP_LEVEL_T>
-struct is_string_type {
-  constexpr static bool value = is_string_type_impl<decay<TOP_LEVEL_T>>::value;
-};
+// Explicitly exclude string-related types from cereal-based implementations.
+// (Had troubles with Microsoft Visual Studio compiler matching too much, likely, correctly. -- D.K.)
```

Comment:
Hmm, I specifically kept it here since it has to do with Cereal, not with strings. Maybe remove it altogether?